### ISSUE TITLE
How to use Chrome profile other than default?

### ISSUE THREAD

**Issue by stefek99** [webdriverio/webdriverio#674]
I'm using a modified example to use an existing Chrome profile:

```
var webdriverio = require('../index');
var options = {
    desiredCapabilities: {
        browserName: 'chrome',
        chromeOptions: {
            args : ["user-data-dir=c:\\Users\\a-miste\\AppData\\Local\\Google\\Chrome\\User Data"]
        }
    }
};

webdriverio
    .remote(options)
    .init()
    .url('http://www.google.com')
    .title().then(function (title) {
        console.log('Title was: ' + title.value);
    });
    //.end(); // DO NOT END SESSION AS WANT TO NAVIGATE TO: chrome://version
```

This documentation part - https://sites.google.com/a/chromium.org/chromedriver/capabilities - is very helpful - *"you can use the user-data-dir Chrome command-line switch to tell Chrome which profile to use"*

The problem is - how do I specify profile other than default?

If I try:
```args : ["user-data-dir=c:\\Users\\a-miste\\AppData\\Local\\Google\\Chrome\\User Data\\Profile 1"]```

And then navigate to ```chrome://version``` the profile is as follows:
![chrome profile](https://cloud.githubusercontent.com/assets/249703/8987423/e96fe71e-36d8-11e5-8c7a-e5b7d7acfd3d.PNG)

I tried and it is not a matter of a trailing slash. A workaround is to copy a non-default profile into some other location and place it under ```default``` folder.

UPDATE: Issue on SeleniumHQ - https://github.com/SeleniumHQ/selenium/issues/854

**Comment by christian-bromann** [comment]
Moved to webdriverio/wdio-chrome-profile-service#1

**Comment by Pasupathi-Rajamanickam** [comment]
@christian-bromann  webdriverio/wdio-chrome-profile-service#1 is a link?

**Comment by Pasupathi-Rajamanickam** [comment]
@christian-bromann I added `'user-data-dir=/tmp/temp_profile'` in my capabilities, created same exact folder and its empty, when my test launches chrome and I was checking `chrome://version/` it still creates some random folder and using that as profile. I tried to use the same old random profile and configured for next test executions, its still creating new folders and profile. Any idea? I also closed all process using `pkill -f 2.38-x64-chromedriver`

**Comment by ValentinCondurache** [comment]
If anyone is struggling to make this work, I am leaving here my solution, maybe it helps. I've lost some time to figure out that we need also another argument (`--profile-directory`) to run tests with a chrome profile:

    runTestWithSpecificChromeProfile: async function (capabilities, profileName) {
        let userDataDir = getChromeUserDataDirPath();
        const fullPath = `${userDataDir}\/${profileName}`;

        Assert.ok(
            fs.existsSync(fullPath),
            `Profile specified (${profileName}) is not existing as a folder in Chrome user data directory (${userDataDir})`,
        );

        // I'm not sure why, but for macOS the userDataDir path should include also profile folder to work
        if (require('os').type() === 'Darwin') userDataDir = fullPath;

        [`--user-data-dir=${userDataDir}`, `--profile-directory=${profileName}`].forEach((arg) =>
            capabilities['goog:chromeOptions'].args.push(arg),
        );
    },
